eureka-client/src/test/java/com/netflix/discovery/converters/wrappers/CodecWrappersTest.java

```diff
@@ -0,0 +1,61 @@
+package com.netflix.discovery.converters.wrappers;
+
+import junit.framework.Assert;
+import org.junit.Test;
+
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * @author David Liu
+ */
+public class CodecWrappersTest {
+
+    private static String testWrapperName = "FOO_WRAPPER";
+
+    @Test
+    public void testRegisterNewWrapper() {
+        Assert.assertNull(CodecWrappers.getEncoder(testWrapperName));
+        Assert.assertNull(CodecWrappers.getDecoder(testWrapperName));
+
+        CodecWrappers.registerWrapper(new TestWrapper());
+
+        Assert.assertNotNull(CodecWrappers.getEncoder(testWrapperName));
+        Assert.assertNotNull(CodecWrappers.getDecoder(testWrapperName));
+    }
+
+    private final class TestWrapper implements EncoderDecoderWrapper {
```

Comment:
Good point. I think we do need an abstraction over our codecs, as even when we standardize on a single codec in the future, there needs to be a way to add additional codecs for extension purposes.

The APIs as they exist today are pretty bare and generic. We can always add additional ones in the future as needed.